protocols/virtualfund/virtualfund-single-hop_test.go

```diff
@@ -171,24 +149,58 @@ func TestSingleHopVirtualFund(t *testing.T) {
 			}
 
 			expectedSignedState := state.NewSignedState(o.V.PreFundState())
-			aliceSig, _ := o.V.PreFundState().Sign(my.privateKey)
-			_ = expectedSignedState.AddSignature(aliceSig)
+			mySig, _ := o.V.PreFundState().Sign(my.privateKey)
+			_ = expectedSignedState.AddSignature(mySig)
 
-			forBob := protocols.Message{To: bob.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}}
-			forIrene := protocols.Message{To: p1.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}}
-			want := protocols.SideEffects{MessagesToSend: []protocols.Message{forIrene, forBob}}
+			want := protocols.SideEffects{MessagesToSend: []protocols.Message{}}
+			switch my.role {
+			case 0:
+				{
+					want.MessagesToSend = append(want.MessagesToSend, protocols.Message{To: p1.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}})
+					want.MessagesToSend = append(want.MessagesToSend, protocols.Message{To: bob.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}})
+
+				}
+			case 1:
+				{
+					want.MessagesToSend = append(want.MessagesToSend, protocols.Message{To: alice.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}})
+					want.MessagesToSend = append(want.MessagesToSend, protocols.Message{To: bob.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}})
+				}
+			case 2:
+				{
+					want.MessagesToSend = append(want.MessagesToSend, protocols.Message{To: alice.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}})
+					want.MessagesToSend = append(want.MessagesToSend, protocols.Message{To: p1.address, ObjectiveId: o.Id(), SignedStates: []state.SignedState{expectedSignedState}})
+				}
+			}
+			// TODO ^^^^ the test is sensitive to the order of the messages. It should not be.
 
 			if diff := cmp.Diff(want, got); diff != "" {
 				if diff := cmp.Diff(want, got); diff != "" {
 					t.Errorf("TestCrank: side effects mismatch (-want +got):\n%s", diff)
 				}
 
 			}
+
 			// Manually progress the extended state by collecting prefund signatures
+			aliceSig, _ := vPreFund.Sign(alice.privateKey)
 			bobSig, _ := vPreFund.Sign(bob.privateKey)
 			p1Sig, _ := vPreFund.Sign(p1.privateKey)
-			o.V.AddStateWithSignature(vPreFund, bobSig)
-			o.V.AddStateWithSignature(vPreFund, p1Sig)
+			switch my.role {
+			case 0:
+				{
```

Comment:
This might be another good candidate for a helper function. It could probably be smart enough to figure out which signatures are missing and add them so we don't have to have a case statement.

Reply:
47fc3b77adf2afd44f04ce75eb0c6d696c0b2b14